### PR TITLE
FIX : Reserved SQL Word not allowed as extrafield

### PR DESCRIPTION
**Pull request by defrance**
Alternative bugfix of https://github.com/Dolibarr/dolibarr/pull/5991
seems better
@atm-florian : can you fill the postgres variant side?

### CODE REVIEW

File: htdocs/core/actions_extrafields.inc.php [hunk 1]
```diff
@@ -127,6 +128,32 @@
     			}
     		}
         }
+		if (! $error)
+		{
+			// attrname must not be a reserved word
+			if ($db->type == 'pgsql')
+			{
+				$sql="SELECT * FROM pg_get_keywords()";
+			}
```
Comment (ghost) [inline htdocs/core/actions_extrafields.inc.php]: $res = $db->query($sql);
-              if ($res)
-                  while ($rec = $db->fetch_array($res)) {
-                      $cats[] = $rec['word'];}

File: htdocs/core/actions_extrafields.inc.php [hunk 2]
```diff
@@ -127,6 +128,32 @@
     			}
     		}
         }
+		if (! $error)
+		{
+			// attrname must not be a reserved word
+			if ($db->type == 'pgsql')
+			{
+				$sql="SELECT * FROM pg_get_keywords()";
```
Comment (ghost) [inline htdocs/core/actions_extrafields.inc.php]: SELECT \* FROM pg_get_keywords() WHERE catcode='R'

**Review by ghost — CHANGES_REQUESTED** [review]

File: htdocs/core/actions_extrafields.inc.php [hunk 3]
```diff
@@ -127,6 +128,28 @@
...
     		}
         }
+		if (! $error)
+		{
+			$cats = array ();
+			// attrname must not be a reserved word
+			if ($db->type == 'pgsql')
+				$sql ="SELECT word as reservedword FROM pg_get_keywords() WHERE catcode='R'";
+			else
+				$sql ="SELECT name as reservedword FROM mysql.help_keyword";
```
Comment (eldy) [inline htdocs/core/actions_extrafields.inc.php:138]: This are system table.
And the database user, if server setup was done correctly won't have access to such tables.
A hardcoded list of common keys should be used even if it does not prevent 100% of cases.

**Comment by eldy** [comment]
It seems this PR was set with "Discussion" or "PR to fix" tag, but no answer nor correction was provided since a long time. So request is automatically closed.
Please reopened and add a comment if you think this is an error.

**Pull request closed without merging**